docs/madr/1-reassign-evolve.md

```diff
@@ -0,0 +1,105 @@
+# Reassign ANT to Evolve Implementation Kickoff on the ArNS Portal
+
+- Status: Proposed
+- Approvers: [Ariel], [Dylan], [Atticus], [Phil], [David]
+- Date: [2025-01-30]
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+Currently to evolve the ANT we use AOS `Eval` - this introduces the problem of
+evaluating what the current version of the code is. If we maintain a variable in
+the state of the process containing a version number, that tells us only what
+that version number is. At any point the user - owner - of the ANT may `Eval`
+new code into the process resulting in changes to the process that may not
+reflect what we expect the version number we assigned to reflect. To overcome
+this we dryrun interactions on APIs and analyze the results to see if the APIs
+perform as expected - this is very inefficient as dry run calls to CU's are
+expensive in networking client side, as well as for the CU.
+
+And so it is proposed to leverage the `Reassign-Name` handler in conjunction
+with `_boot` to fork the process state to a whole new ANT, assign that new ANT
+as the registered ANT to the specified ArNS name in the registry, and using the
+module ID of the process to identify process version and capabilities, and
+solely the module ID.
```

Comment:
`With the ARIO contract now support `Reassign-Name` and the recent additions to add `_boot` handler...`

add add links referencing these changes (the PRs are fine)

Reply:
[e55f7a1](https://github.com/ar-io/ar-io-ant-process/pull/74/commits/e55f7a1f9a963d263bf50cf267d8e2dc28d3921d)